lib/src/firebase_performance_dio.dart

```diff
@@ -16,12 +16,14 @@ class DioFirebasePerformanceInterceptor extends Interceptor {
   DioFirebasePerformanceInterceptor({
     this.requestContentLengthMethod = defaultRequestContentLength,
     this.responseContentLengthMethod = defaultResponseContentLength,
+    this.generateKey,
   });
 
   /// key: requestKey hash code, value: ongoing metric
-  final _map = <int, HttpMetric>{};
+  final _map = <dynamic, HttpMetric>{};
   final RequestContentLengthMethod requestContentLengthMethod;
   final ResponseContentLengthMethod responseContentLengthMethod;
+  final dynamic Function(Map<String, dynamic>)? generateKey;
```

Comment:
Same as above

Reply:
This will generate a key to save in the Map above. Since the default implementation generates int, but the user may want to generate String or other type, I defined it as dynamyc. But it should not be null...I might have to rethink this. 
In my opinion the most propriatee type would be String for keys. Then I will have to change more the default implementation.